sdk-core/src/main/java/care/data4life/sdk/tag/TagEncoding.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2020 D4L data4life gGmbH / All rights reserved.
+ *
+ * D4L owns all legal rights, title and interest in and to the Software Development Kit ("SDK"),
+ * including any intellectual property rights that subsist in the SDK.
+ *
+ * The SDK and its documentation may be accessed and used for viewing/review purposes only.
+ * Any usage of the SDK for other purposes, including usage for the development of
+ * applications/third-party applications shall require the conclusion of a license agreement
+ * between you and D4L.
+ *
+ * If you are interested in licensing the SDK for your own applications/third-party
+ * applications and/or if you’d like to contribute to the development of the SDK, please
+ * contact D4L by email to help@data4life.care.
+ */
+package care.data4life.sdk.tag
+
+import care.data4life.sdk.lang.D4LException
+import care.data4life.sdk.lang.DataValidationException
+import care.data4life.sdk.wrapper.URLEncoding
+import java.util.Locale
+
+object TagEncoding : TaggingContract.Encoding {
+    @Throws(D4LException::class)
+    private fun validateTag(tag: String) {
+        if (tag.isBlank()) {
+            throw DataValidationException.AnnotationViolation(
+                "Annotation is empty."
+            )
+        }
+    }
+
+    @Throws(D4LException::class)
+    override fun encode(
+        tag: String
+    ): String = URLEncoding.encode(normalize(tag)).toLowerCase(Locale.US)
```

Comment:
Jap it is, otherwise it takes the default of the System, which is not desirable, since we have no control over that.
Yes, I'll add it to the Contract.